osu.Game/Database/RealmContextFactory.cs

```diff
@@ -156,7 +168,9 @@ private void flushContexts()
             Logger.Log(@"Flushing realm contexts...", LoggingTarget.Database);
 
             var previousContext = context;
```

Comment:
Shouldn't this also be in the lock?

Reply:
The assumption is that flush is always run from the update thread (or in a context where updates aren't running) so it's not required. But if it makes you feel better it can be in the lock for sure.

Reply:
In addition, at this point we already have the `blockingWait` lock which would be required to create a new context.

Reply:
Then I don't understand what these changes are fixing, if everything's supposedly single threaded across `Context_get()`, `flushContexts()` and `Update()`. Am I missing something...?

Reply:
in the `update` method the context fetch and `Refresh` call could have a 5 second delay between them. during this time a flush could occur.

Reply:
That would imply that this isn't always run from the update thread, contradictory to what you said, no?

Reply:
Yep, I guess that's correct. Either way the lock is not required for the other reason stated. But I've added it for visual clarity.